### PR TITLE
implement jax.numpy.interp

### LINKED ISSUE #3927: [FR] jnp.interp & jnp.piecewise
Hey, I am trying to create a learning reate schedule and it would be super helpful to have `interp` or `piecewise` for this task.

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #3927; related to #3860

### CODE REVIEW

**Comment by jakevdp** [comment]
GPU/TPU tests pass

**Review by mattjj — APPROVED** [review]
LGTM, nice!

**Comment by AdrienCorenflos** [comment]
@jakevdp, did you see my comment above? Shouldn't be of any harm but am I missing something?

**Comment by jakevdp** [comment]
I don't see any comments on this PR. What are your concerns?

**Comment by AdrienCorenflos** [comment]
Not a concern per se, just a redundancy in the code. Copied the original text below (weird you can't see it)

`x == xp[i]` seems redundant as in that case delta will be equal to dx. What problem does it solve?

**Comment by jakevdp** [comment]
Oh, what I had in mind was efficiency and floating point accuracy. No need to compute `delta * df / dx` for entries where you know the answer *a priori*.

**Comment by AdrienCorenflos** [comment]
Hmmm, re the first point I thought there was no lazy evaluation happening with where (contrarily to lax.cond).

In terms of floating point precision I think you should anyway do (delta/dx)*df instead of delta*(df/dx) as delta and dx should be of comparable value contrarily to df and dx (plus df and dx can have mixed types in all generality).

**Comment by jakevdp** [comment]
> re the first point I thought there was no lazy evaluation happening with where (contrarily to lax.cond).

My understanding is that XLA computation effectively makes these operations lazy.

>  I think you should anyway do (delta/dx)*df instead of delta*(df/dx)

Good point! I'll send a followup.

**Comment by AdrienCorenflos** [comment]
It would be a really good surprise for me if you told me that Jax was not strict tbh. I still work under my tensorflow assumptions of strict evaluation...
